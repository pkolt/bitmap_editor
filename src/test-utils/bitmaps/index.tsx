import { BitmapEntity } from '@/utils/bitmap/types';

export const bitmapEntity: BitmapEntity = {
  id: 'eddc9bc7-b691-46e9-b2b0-be4d32707968',
  name: 'Test',
  createdAt: Date.now(),
  updatedAt: Date.now(),
  data: [4086534384, 252694479],
  width: 8,
  height: 8,
  favorite: false,
};
