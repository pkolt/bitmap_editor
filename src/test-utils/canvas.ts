export class FakeCanvas {
  public width = 0;
  public height = 0;
  #imageData?: ImageData;

  constructor(imageData?: ImageData) {
    this.#imageData = imageData;
  }

  getContext() {
    return {
      fillRect() {},
      clearRect() {},
      drawImage() {},
      beginPath() {},
      moveTo() {},
      lineTo() {},
      stroke() {},
      getImageData: () => this.#imageData,
    };
  }
}
