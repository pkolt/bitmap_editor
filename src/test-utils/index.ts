// Re-export
export * from 'vitest';
// Re-export
export * from '@testing-library/react';

export { renderElement as renderComponent } from './render/renderElement';
export { renderPage } from './render/renderPage';
