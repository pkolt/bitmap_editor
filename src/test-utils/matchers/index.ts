import { expect } from 'vitest';

const isObject = (value: unknown): value is object => {
  return typeof value === 'object' && value !== null && !Array.isArray(value);
};

const isDisable = (elem: unknown): boolean => {
  return isObject(elem) && 'disabled' in elem && !!elem.disabled;
};

const errorDisabled = 'Expected: "enabled", received: "disabled"';
const errorEnabled = 'Expected: "disabled", received: "enabled"';

export const setupMatchers = () => {
  expect.extend({
    toBeEnabled: function (received) {
      const { isNot } = this;
      return {
        pass: !isDisable(received),
        message: () => (isNot ? errorEnabled : errorDisabled),
      };
    },
    toBeDisabled: function (received) {
      const { isNot } = this;
      return {
        pass: isDisable(received),
        message: () => (isNot ? errorDisabled : errorEnabled),
      };
    },
  });
};
