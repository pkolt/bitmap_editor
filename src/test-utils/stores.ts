import { useBitmapStore } from '@/stores/bitmaps';
import { useSettingsStore } from '@/stores/settings';
import { afterEach } from 'vitest';
import { act } from '@testing-library/react';

export const setupStores = () => {
  afterEach(() => {
    act(() => {
      useBitmapStore.getState().reset();
      useSettingsStore.getState().reset();
    });
  });
};
