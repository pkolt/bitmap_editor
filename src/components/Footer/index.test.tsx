import { Footer } from '.';
import { render, expect, test } from '@/test-utils';

test('show component', () => {
  const { container } = render(<Footer />);
  expect(container).matchSnapshot();
});
