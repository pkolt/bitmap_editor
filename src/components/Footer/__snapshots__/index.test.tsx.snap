// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`show component 1`] = `
<div>
  <footer
    class="container-fluid p-3 bg-light text-center rounded-top-2"
  >
    Created by 
    <a
      href="https://github.com/pkolt"
    >
      Pavel Koltyshev
    </a>
     Â© 2023-
    2024
  </footer>
</div>
`;
